consensus/istanbul/backend/handler.go

```diff
@@ -286,3 +301,154 @@ func (sb *Backend) UnregisterPeer(peer consensus.Peer, isProxiedPeer bool) {
 		}
 	}
 }
+
+// Handshake allows the initiating peer to identify itself as a validator
+func (sb *Backend) Handshake(peer consensus.Peer) (bool, error) {
+	// Only written to if there was a non-nil error when sending or receiving
+	errCh := make(chan error)
+	isValidatorCh := make(chan bool)
+
+	sendHandshake := func() {
+		var msg *istanbul.Message
+		var err error
+		peerIsValidator := peer.PurposeIsSet(p2p.ValidatorPurpose)
+		if peerIsValidator {
+			// msg may be nil
+			msg, err = sb.retrieveEnodeCertificateMsg()
+			if err != nil {
+				errCh <- err
+				return
+			}
+		}
+		// Even if we decide not to identify ourselves,
+		// send an empty message to complete the handshake
+		if msg == nil {
+			msg = &istanbul.Message{}
+		}
+		msgBytes, err := msg.Payload()
+		if err != nil {
+			errCh <- err
+			return
+		}
+		err = peer.Send(istanbulValidatorHandshakeMsg, msgBytes)
+		if err != nil {
+			errCh <- err
+		}
+		isValidatorCh <- peerIsValidator
+	}
+	readHandshake := func() {
+		isValidator, err := sb.readValidatorProofMessage(peer)
```

Comment:
Regarding naming conventions, seems like `readValidatorProofMessage` should be renamed to `readIstanbulValidatorHandshakeMsg`.

Reply:
Oops, thanks